qiskit_experiments/randomized_benchmarking/interleaved_rb_analysis.py

```diff
@@ -89,51 +81,98 @@ def fit_fun_interleaved(x, a, _, alpha_int, b):
-            ax = plotting.plot_curve_fit(fit_fun_interleaved, analysis_result, ax=ax)
-            ax = plotting.plot_scatter(std_xdata, std_ydata, ax=ax)
-            ax = plotting.plot_scatter(int_xdata, int_ydata, ax=ax)
-            ax = plotting.plot_errorbar(std_xdata, std_ydata, std_ydata_sigma, ax=ax)
-            ax = plotting.plot_errorbar(int_xdata, int_ydata, int_ydata_sigma, ax=ax)
+
+        if plot and plotting.HAS_MATPLOTLIB:
+            ax = plotting.plot_curve_fit(fit_fun_standard, analysis_result, ax=ax, color="blue")
+            ax = plotting.plot_curve_fit(
+                fit_fun_interleaved,
+                analysis_result,
+                ax=ax,
+                color="green",
+            )
+            ax = self._generate_multi_scatter_plot(series_raw, x_raw, y_raw, ax=ax)
+            ax = self._generate_multi_errorbar_plot(series, xdata, ydata, ydata_sigma, ax=ax)
             self._format_plot(ax, analysis_result)
-            analysis_result.plt = plotting.pyplot
+            ax.legend(loc="center right")
+            figures = [ax.get_figure()]
+        else:
+            figures = None
+        return analysis_result, figures
+
+    @staticmethod
+    def _generate_multi_scatter_plot(series, xdata, ydata, ax):
+        """Generate scatter plot of raw data"""
+        idx0 = series == 0
+        idx1 = series == 1
+        ax = plotting.plot_scatter(xdata[idx0], ydata[idx0], ax=ax)
+        ax = plotting.plot_scatter(xdata[idx1], ydata[idx1], ax=ax)
+        return ax
 
-        return analysis_result, None
+    @staticmethod
+    def _generate_multi_errorbar_plot(series, xdata, ydata, sigma, ax):
+        """Generate errorbar plot of average data"""
+        idx0 = series == 0
+        idx1 = series == 1
+        ax = plotting.plot_errorbar(
+            xdata[idx0],
+            ydata[idx0],
+            sigma[idx0],
+            ax=ax,
+            label="Standard",
+            marker=".",
+            color="red",
+        )
+        ax = plotting.plot_errorbar(
+            xdata[idx1],
+            ydata[idx1],
+            sigma[idx1],
+            ax=ax,
+            label="Interleaved",
+            marker="^",
+            color="orange",
+        )
+        return ax
+
+    @staticmethod
+    def _p0_multi(series, xdata, ydata, num_qubits):
+        """Initial guess for the fitting function"""
+        std_idx = series == 0
+        p0_std = RBAnalysis._p0(xdata[std_idx], ydata[std_idx], num_qubits)
+        int_idx = series == 1
+        p0_int = RBAnalysis._p0(xdata[int_idx], xdata[int_idx], num_qubits)
+        return {
+            "a": np.mean([p0_std["a"], p0_int["a"]]),
+            "alpha": p0_std["alpha"],
+            "alpha_c": p0_int["alpha"] / p0_std["alpha"],
+            "b": np.mean([p0_std["b"], p0_int["b"]]),
+        }
 
     @classmethod
     def _format_plot(cls, ax, analysis_result, add_label=True):
```

Comment:
Do you think this is needed now? In old Ignis implementation this is kind of necessary because fit values are protected member. But now these values can be seen in the repr of analysis result object.

Reply:
I think it's nice to keep the info in the figure so you can get the main details without having to also look at the result.